src/processor/reconnection-graph.service.ts

```diff
@@ -314,7 +315,12 @@ export class ReconnectionGraphService {
   ): Promise<ConnectAction[]> {
     const dsnpKeys: DsnpKeys = await this.formDsnpKeys(dsnpUserId);
     const actions: ConnectAction[] = [];
-    // this.logger.debug(`Graph connections for user ${dsnpUserId.toString()}: ${JSON.stringify(graphConnections)}`);
+    const delegationResult: Option<Vec<SchemaGrantResponse>> = await this.blockchainService.rpc('msa', 'grantedSchemaIdsByMsaId', dsnpUserId, providerId);
+    if (delegationResult.isNone) {
+      this.logger.log(`User ${dsnpUserId} has no delegations to provider ${providerId}`);
+      return actions;
+    }
+    const delegations = delegationResult.unwrap().toArray();
```

Comment:
This file contains the only code change to the service. Originally we were querying the RPC to get the current user's delegations inside the closure for each connection; here we've refactored it so we only need to call it once. We still can't avoid the RPC call to get schema-specific delegations for each connection, but in my test case of 7k connections, this reduced RPC calls here from 14k -> 7,001.